ansible/roles/signal/tasks/main.yml

```
@@ -0,0 +1,49 @@
+---
+- pause:
+    prompt: "{{ lookup('template', 'enter_captcha.j2') | trim }}"
+  register: captcha_prompt_result
+
+- set_fact:
+    captcha: "{{ captcha_prompt_result.user_input }}"
+
+- name: Register phone number
+  uri:
+    url: "{{ signal_rest_cli_endpoint }}/v1/register/{{ signal_server_phone_number }}"
+    method: POST
+    status_code: 201
+    body_format: json
+    body:
+      captcha: "{{ captcha }}"
+
+- pause:
+    prompt: "{{ lookup('template', 'enter_verification_token.j2') | trim }}"
+  register: verification_token_prompt_result
+
+- set_fact:
+    verification_token: "{{ verification_token_prompt_result.user_input }}"
+
+- name: Verify phone number
+  uri:
+    url: "{{ signal_rest_cli_endpoint }}/v1/register/{{ signal_server_phone_number }}/verify/{{ verification_token }}"
+    method: POST
+    status_code: 201
+    body_format: json
+    body:
+      ping: "string"
+
+- name: Query id of app container
+  command:
+    cmd: docker-compose -f docker-compose.yml -f docker-compose.prod.yml ps -q app
+    chdir: /home/ansible
+  register: get_container_id
+
+- name: Save container id of app
+  set_fact:
+    app_container_id: "{{ get_container_id.stdout }}"
+
+- name: Setup cronjob to receive messages
+  ansible.builtin.cron:
+    name: "Receive signal messages"
+    minute: "*" # job gets run every minute
+    job: "cd {{ ansible_facts.env.HOME }} && docker exec -it {{ app_container_id }} bin/rails signal:receive_messages"
```

As the playbook is only run once during set up, this fails as soon as the container ID changes (e.g. when the container is recreated).

```suggestion
    job: "cd /home/ansible && docker-compose -f docker-compose.yml -f docker-compose.prod.yml exec app bin/rails signal:receive_message"
```

---

Also, this currently adds a job to the root user’s crontab, we might want to change that!

---

Okay, even with these changes, when executed by cron, the rake task isn’t run for some reason. I’ve got no time to debug this right now. Any ideas @phoeinx @roschaefer?

* It works when I invoke it manually, so I thought it might be related to the environment?

---

Oh, that makes a lot of sense, of course, and changing the playbook as you suggested makes sense to me.
Regarding the still not run rake task, I wonder why you changed `signal:receive_messages` to `signal:receive_message`? I just checked and on the latest state on this branch, the rake task is still called `receive_messages`.

---

@phoeinx That’s a typo in the suggestion, sorry about the confusion. (I’m pretty sure I set up the job with the correct rake task, but maybe double-check that.)

---

@phoeinx and I isolated the issue to the `docker-compose` command itself. It seems to work for me now when I use absolute paths not only for the `docker-compose` command, but also for the `docker-compose.yml` files. (Changing to `/home/ansible` before executing the command wasn’t enough.)

---

Additionally, we need to set the `-T` flag for the `exec` command. From the docs:

> Disable pseudo-tty allocation. By default `docker-compose exec` allocates a TTY.